### DISCUSSION TITLE
Analytics on Attestation

### DISCUSSION THREAD

**Discussion by realchoubey** [celo-org/social-connect#231] [News (Updates)]
What are the some major matrices which will be useful in terms of Analytics on Attestations?

**Comment by realchoubey** [comment]
@critesjosh @carterqw2 @eelanagaraj @0xarthurxyz 

We are making subgraph and explorer to track key attestation metrics. 
- Attestation request daily/weekly/monthly volume
- Attestation completed daily/weekly/monthly volume
- Total identifiers 
- Fee of an attestation: Useful in term of avg. attestation fee or to see trend over time.
- Issuer details related to how many of attestation completed by an issuer
- Issuer wait block: it is usually 4 block but in case it increase we can have trend to be shown over time.

What other metrics will be useful to track in subgraph and explorer?

**Reply by eelanagaraj** [reply]
I think these would be helpful! Attestations completed by issuers would be useful. One other interesting thing to look at per issuer is completion/incompletion of an attestation conditional on whether the other attestations were completed in the same verification attempt. (In other words: for one verification attempt, 3 attestations are requested and assigned to issuers A,B, and C. It would be great to understand when A is the only issuer that missed an attestation, as this could be an indication that delivery should in theory be possible, especially if this continually happens for an issuer).

One question on `fee of an attestation` -- are you including gas fees paid in this or just the actual AttestationRequestFee?

Looking forward to seeing your metrics on this! 🙏

**Reply by critesjosh** [reply]
+1 to everything that @eelanagaraj said.

it might be interesting to look at how many times an attestation issuer has been selected, if that's possible. this could help provide insight about completion rates by comparing how many times they've been selected to how many attestations have been completed by them.

**Reply by realchoubey** [reply]
Thanks @eelanagaraj and @critesjosh for you advice, we will check the feasibility of suggestions.

@eelanagaraj, we just showing actual AttestationRequestFee, Would it be good to show `gas fees paid`?

**Reply by eelanagaraj** [reply]
Hey @realchoubey! I think showing gas fees could be more interesting since the `AttestationRequestFee` shouldn't actually change very often I believe 🤔 
From a quick check of the logs for the `AttestationProxy` contract, it looks like the `AttestationRequestFeeSet` event has [only been emitted once](https://explorer.celo.org/api/?module=logs&action=getLogs&fromBlock=0&toBlock=12452774&address=0xdC553892cdeeeD9f575aa0FBA099e5847fd88D20&topic0=0x7cf8b633f218e9f9bc2c06107bcaddcfee6b90580863768acdcfd4f05d7af394), so I'd expect this value to be quite stable! Are you seeing changes in the fees?

**Reply by realchoubey** [reply]
@eelanagaraj, We checked and come up with what we can show as per requested. Could you please confirm once this would work or you suggest something else?
So basically we are thinking of having following attributes
```
Identifier
Account
No of issuer selected
No of success attestation
No of failure attestation (Did you mean to show issuer address here?)
```
So `Identifier and account` will be used to identify uniqueness. 


Regrading your question, yes you are right there is only single event. We will look forward toward considering attestation gas fee paid.

**Reply by realchoubey** [reply]
@eelanagaraj, any thought on my comment?

**Reply by realchoubey** [reply]
@eelanagaraj @critesjosh 
We have observed that for same pair of identifier and account Requested attestations there are multiple issuer selected and all comes to completed Attestation which is resulting in more completed attestation as compared to requested attestation.
My question is while showing total number of completed attestations, should we consider all completed attestation with different issuer for same pair for identifier and account or just a single count for pair of identifier and account ignoring different issuers for same attestation request?

**Reply by eelanagaraj** [reply]
Hey @realchoubey sorry for the delayed response, missed the earlier notification 🤔 I think those stats make sense to track!

For your question: one other thing to note is that the `AttestationsRequested` event should also include the number of `attestationsRequested`. You could track both the event & number of attestations requested per account, which I believe should be >= # completed per account, or are you seeing something different there? I think it would generally be helpful to have an understanding of completion for (identifier, account, issuer), so we can understand which issuers are performing better/worse!

**Comment by arthurgousset** [comment]
Quick ping because I transferred this discussion 

- from: `celo-org/celo-monorepo/discussions`
- to: `celo-org/identity/discussions`

to be more flexible in the types of discussions we can host:

- 💡 [Feature requests (Feedback)](https://github.com/celo-org/identity/discussions/categories/feature-requests-feedback)
- 😺 [Help needed (Q&A)](https://github.com/celo-org/identity/discussions/categories/help-needed-q-a)
- 🗳️ [Polls (Voting)](https://github.com/celo-org/identity/discussions/categories/polls-voting)
- 🙌 [Show and tell (Updates)](https://github.com/celo-org/identity/discussions/categories/show-and-tell-updates)
 
Sorry for any inconvenience @realchoubey @critesjosh @eelanagaraj 🙏

**Comment by realchoubey** [comment]
@eelanagaraj  @critesjosh  @0xarthurxyz 
The below attestation dashboard are live, please find details on [celo forum](https://forum.celo.org/t/attestation-dashboard-and-analytics/3674).

Dashboards includes:
- Attestation
- Identifier 
- Issuer

Please let me know about your feedback.

**Comment by arthurgousset** [comment]
Hi @realchoubey, 

@eelanagaraj and team are working on updates in [Escrow.sol](https://github.com/celo-org/celo-monorepo/blob/6b6ce69fde8f4868b54abd8dd267e5313c3ddedd/packages/protocol/contracts/identity/Escrow.sol), which we wanted to make you aware of ahead of time in case you are indexing its events (there is no immediate rush):

The changes: 

1. add support for federated attestations ([FederatedAttestations.sol](https://github.com/celo-org/celo-monorepo/blob/ASv2/packages/protocol/contracts/identity/FederatedAttestations.sol))
2. fix a https://github.com/celo-org/celo-monorepo/pull/9617
3. fix an inconsistency in the data emitted by the [Withdrawal event](https://github.com/celo-org/celo-monorepo/blob/6b6ce69fde8f4868b54abd8dd267e5313c3ddedd/packages/protocol/contracts/identity/Escrow.sol#L34) 👈 **Topic of this thread**

You can find more details here: 
- https://github.com/celo-org/identity/discussions/25
